internal/controller/api/v1alpha1/host_types.go

```diff
@@ -17,25 +17,188 @@ limitations under the License.
+	SerialNumber string `json:"serialNumber,omitempty"`
+	ProductName  string `json:"productName,omitempty"`
+	Manufacturer string `json:"manufacturer,omitempty"`
+	Virtual      bool   `json:"virtual,omitempty"`
+}
+
+type HostInventory struct {
+	// Name in REST API: timestamp
+	ReportTime   *metav1.Time     `json:"reportTime,omitempty"`
+	Hostname     string           `json:"hostname,omitempty"`
+	BmcAddress   string           `json:"bmcAddress,omitempty"`
+	BmcV6address string           `json:"bmcV6Address,omitempty"`
+	Memory       HostMemory       `json:"memory,omitempty"`
+	Cpu          HostCPU          `json:"cpu,omitempty"`
+	Interfaces   []HostInterface  `json:"interfaces"`
+	Disks        []HostDisk       `json:"disks"`
+	Boot         HostBoot         `json:"boot,omitempty"`
+	SystemVendor HostSystemVendor `json:"systemVendor,omitempty"`
+}
 
 // HostSpec defines the desired state of Host
 type HostSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// The name of the Cluster CR
+	ClusterRef              *corev1.ObjectReference `json:"clusterRef"`
+	Role                    models.HostRole         `json:"role" protobuf:"bytes,1,opt,name=role,casttype=HostRole"`
+	Hostname                string                  `json:"hostname,omitempty"`
+	MachineConfigPool       string                  `json:"machineConfigPool,omitempty"`
+	Enabled                 *bool                   `json:"enabled,omitempty"`
+	IgnitionConfigOverrides string                  `json:"ignitionConfigOverrides,omitempty"`
+	InstallerArgs           string                  `json:"installerArgs,omitempty"`
+}
+
+type HardwareValidationInfo struct {
+	HasInventory       host.ValidationStatus `json:"hasInventory,omitempty"`
+	HasMinCPUCores     host.ValidationStatus `json:"hasMinCPUCores,omitempty"`
+	HasMinMemory       host.ValidationStatus `json:"hasMinMemory,omitempty"`
+	HasMinValidDisks   host.ValidationStatus `json:"hasMinValidDisks,omitempty"`
+	HasCpuCoresForRole host.ValidationStatus `json:"hasCPUCoresForRole,omitempty"`
+	HasMemoryForRole   host.ValidationStatus `json:"hasMemoryForRole,omitempty"`
+	IsHostnameValid    host.ValidationStatus `json:"isHostnameValid,omitempty"`
+	IsHostnameUnique   host.ValidationStatus `json:"isHostnameUnique,omitempty"`
+	IsPlatformValid    host.ValidationStatus `json:"isPlatformValid,omitempty"`
+}
+
+type NetworkValidationInfo struct {
+	IsConnected            host.ValidationStatus `json:"isConnected,omitempty"`
+	IsMachineCidrDefined   host.ValidationStatus `json:"isMachineCIDRDefined,omitempty"`
+	BelongsToMachineCidr   host.ValidationStatus `json:"belongsToMachineCIDR,omitempty"`
+	IsAPIVipConnected      host.ValidationStatus `json:"isAPIvipConnected,omitempty"`
+	BelongsToMajorityGroup host.ValidationStatus `json:"belongsToMajorityGroup,omitempty"`
+	IsNTPSynced            host.ValidationStatus `json:"isNTPSynced,omitempty"`
```

Comment:
referencing [this comment](https://github.com/openshift/assisted-service/pull/861#discussion_r551468404), which github won't let me reply to for some reason, what the plan for all of these prefixes? I understood @avishayt 's sentiment to be that we should change these to match the k8s convention and also change the REST API to match?

Reply:
Yes

Reply:
Done